### PR TITLE
UI changes for set up payments task

### LINKED ISSUE #32131: Payments task: UI changes to the additional payment gateways section
**Description:** we want to introduce some tweaks to the additional payment gateways section, in order to reduce cognitive load and increase the visibility of WCPay.

**Acceptance criteria:**
- [x] Split the section into two sections: "Choose a payment provider" [online payments] and "Offline payment methods" [offline payments]
- [x] Change the CTA copy to "Get started"
- [x] Update the payment's icon
- [x] Add a "See more" option at the bottom of the online payments section with a link to https://woocommerce.com/product-category/woocommerce-extensions/payment-gateways/online-payments/

If the store is located in a country where WCPayments is eligible but it's not installed
- [x] Hide the online and offline payment sections under an "Other payment methods" toggle
- [x] Remove the online payment header section

**Event tracking:**
We already have most of the tracking in place. We should only add new event Tracking to the following actions:
- Show/Hide the "Other payment methods" section
- Click on the "See more" option

Ilyas' edit:

I've moved the conditional hiding of "See more" and additional gateway acceptance criteria to a more relevant issue, https://github.com/woocommerce/woocommerce/issues/32132

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Closes #32131.

This PR accomplishes the following ACs:
- Split the section into two sections: "Choose a payment provider" [online payments] and "Offline payment methods" [offline payments]
- Change the CTA copy to "Get started"
- Update the payment's icon
- Add a "See more" option at the bottom of the online payments section with a link to https://woocommerce.com/product-category/woocommerce-extensions/payment-gateways/online-payments/
- Removes "Enabled payment gateways" section

If the store is located in a country where WCPayments is eligible but it's not installed
-  Hide the online and offline payment sections under an "Other payment methods" toggle
- Remove the online payment header section

Adds the following event tracking:
- `wcadmin_tasklist_payment_see_more` when "See more" button is clicked
- `wcadmin_tasklist_payment_show_toggle` when "Other payment method" button is clicked

### How to test the changes in this Pull Request:

**Test with eligible WCPay**:
1. In a fresh site, go through the setup wizard with WCPay eligible country but don't install WCpay.
2. Set `woocommerce_show_marketplace_suggestions` to `no`
3. Go to "Set up payments" task
4. Observe only WCPay card is displayed with an "Other payment methods" button
5. Click on "Other payment methods" toggle button
6. Observe the hidden payment methods are shown

image

**Test with ineligible WCPay**
1. Set store country to somewhere ineligible for WCPay, such as Mexico.
2. Go to "Set up payments" task
3. Observe the "Choose a payment provider" section and "Offline payment methods" section
4. Observe no "Other payment methods" toggle button

image

**Test various UI changes**
1. Go to "Set up payments" task
2. Observe the following:
    - Payment method icons are now smaller
    - "Set up" button next to methods has been changed to "Get started"

**Test new tracks**
1. Set store country to somewhere eligible for WCPay, such as US.
1. Go to "Set up payments" task
2. Open up browser console and run `localStorage.setItem( 'debug', 'wc-admin:tracks' );` to set up tracks debugging
3. Click on "Other payment methods" button
4. Observe `wcadmin_tasklist_payment_show_toggle` track is recorded with `toggle` and `payment_method_count` props
5. Click on "See more" button
6. Observe `wcadmin_tasklist_payment_see_more` track is recorded
7. Observe the "See more" button should go to `https://woocommerce.com/product-category/woocommerce-extensions/payment-gateways/?utm_source=payments_recommendations`

### Changelog entry

Dev: UI changes for set up payments task

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/Action.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ export const Action = ( {
 	markConfigured,
 	onSetUp = () => {},
 	onSetupCallback,
-	setupButtonText = __( 'Set up', 'woocommerce' ),
+	setupButtonText = __( 'Get started', 'woocommerce-admin' ),
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/Action.js]: Is this intended that the text domain is set to `woocommerce-admin`?

File: plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/Setup/Configure.js [hunk 2]
```diff
@@ -156,7 +156,7 @@ export const Configure = ( { markConfigured, paymentGateway } ) => {
 				</p>
 			) }
 			<Button isPrimary href={ settingsUrl }>
-				{ __( 'Set up', 'woocommerce' ) }
+				{ __( 'Get started', 'woocommerce-admin' ) }
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/Setup/Configure.js]: Also, 'woocommerce-admin' text domain here.

File: plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/List/List.js [hunk 3]
```diff
@@ -30,6 +31,11 @@ export const List = ( {
 					/>
 				);
 			} ) }
+			{ footerLink ? (
+				<CardFooter isBorderless>{ footerLink }</CardFooter>
+			) : (
+				''
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/List/List.js]: Just curious. Why do you use an empty string here but `null` in line 22? Are there any rules?

Comment (chihsuan) [reply plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/List/List.js]: I think we also use `&&` somewhere.

File: plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/Toggle/Toggle.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+				) : (
+					<ChevronDownIcon size={ 18 } />
+				) }
+			</Button>
+			{ isShow ? children : null }
+		</div>
+	);
+};
+
+// export default Toggle;
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/Toggle/Toggle.js]: Could we remove this?

File: plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/test/index.js [hunk 5]
```diff
@@ -211,4 +245,59 @@ describe( 'PaymentGatewaySuggestions', () => {
 			selected: 'ppcp_gateway',
 		} );
 	} );
+
+	test( 'should record event correctly when other payment methods is clicked', () => {
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/test/index.js:249]: Thanks for adding tests for track events. 👍

File: plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/Action.js (hunk 1 above)
Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/Action.js]: I think there was a merge conflict, thanks for catching this!

File: plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/List/List.js (hunk 3 above)
Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions/components/List/List.js]: I agree its simpler with just `&&`, thanks!

**Comment by ilyasfoo** [comment]
@chihsuan Thanks for the super quick comments so far, I've addressed all of them in [0b2df9a](https://github.com/woocommerce/woocommerce/pull/32467/commits/0b2df9a95657456c2e03dece9e91cb2bc3d7c390)

File: packages/js/data/src/plugins/types.ts [hunk 6]
```diff
@@ -39,6 +39,7 @@ export type Plugin = {
 	recommendation_priority?: number;
 	is_visible?: boolean;
 	is_local_partner?: boolean;
+	is_offline?: boolean;
```
Comment (chihsuan) [inline packages/js/data/src/plugins/types.ts:42]: Shall we simply add a changelog entry for `data` package?

**Review by chihsuan — DISMISSED** [review]
Thanks for addressing comments quickly! Another comment for changelog. Others look good and tested well. Nice work. 💯 

See if @adrianduffell like to do a second review. 😃

**Comment by adrianduffell** [comment]
> See if @adrianduffell like to do a second review. 

Feel free to merge and I'll take a look at it on `trunk` later 😃

**Review by chihsuan — DISMISSED** [review]

**Comment by chihsuan** [comment]
Oops, looks like the e2e action is failing in the last commit. 👀

**Comment by ilyasfoo** [comment]
@chihsuan Thanks! I've fixed e2e tests in [859ab88](https://github.com/woocommerce/woocommerce/pull/32467/commits/859ab8884b4b69d98e1a06927e4c88b313b14fb7)

File: packages/js/data/src/plugins/types.ts (hunk 6 above)
Comment (ilyasfoo) [reply packages/js/data/src/plugins/types.ts:42]: Sounds good! Added in [1fb2418](https://github.com/woocommerce/woocommerce/pull/32467/commits/1fb2418e57d553a3f607b904e9ddf35da1c8d85e)

**Review by chihsuan — APPROVED** [review]
Looks good! Thanks for updating e2e tests as well. 💯

**Pull request merged**